Client/Application/AppDelegate.swift

```diff
@@ -135,6 +142,52 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
         SDWebImageDownloader.sharedDownloader().setValue(firefoxUA, forHTTPHeaderField: "User-Agent")
     }
+
+    func application(application: UIApplication, handleActionWithIdentifier identifier: String?, forLocalNotification notification: UILocalNotification, completionHandler: () -> Void) {
+        if let actionId = identifier {
+            if let action = SentTabAction(rawValue: actionId) {
+                switch(action) {
+                case .View:
+                    viewURLInNewTab(notification)
+                    break
+                case .Bookmark:
+                    addBookmark(notification)
+                    break
+                case .ReadingList:
```

Comment:
So, I didn't remove these actions, but I have ensured that the tab is opened and then added to bookmarks/reading list, so you are not just looking at FF with an old view, but at the tab you've just added.